add sql statement about pump and drainer (#2945)
pingcap/docs: sql-statements/sql-statement-change-pump.md

---
title: CHANGE PUMP
summary: An overview of the usage of CHANGE PUMP for the TiDB database.
category: reference
---

# CHANGE PUMP

The `CHANGE PUMP` statement modifies the status information for Pump in the cluster.

> **Tip:**
>

> > Pump's state is automatically reported to PD during running, and the statement is used to modify the state information stored in PD only when the state under abnormal circumstances to be inconsistent with the state information stored in PD.

Reviewer: ```suggestion
> Pump's state is automatically reported to PD while running. Only when Pump is under abnormal circumstances and its state is inconsistent with the state information stored in PD, you can use the `CHANGE PUMP` statement to modify the state information stored in PD.
```

## Examples

{{< copyable "sql" >}}

```sql
SHOW PUMP STATUS;
```

```sql
+--------|----------------|--------|--------------------|---------------------|
| NodeID |     Address    | State  |   Max_Commit_Ts    |    Update_Time      |
+--------|----------------|--------|--------------------|---------------------|
| pump1  | 127.0.0.1:8250 | Online | 408553768673342237 | 2019-04-30 00:00:01 |
+--------|----------------|--------|--------------------|---------------------|
| pump2  | 127.0.0.2:8250 | Online | 408553768673342335 | 2019-05-01 00:00:02 |
+--------|----------------|--------|--------------------|---------------------|
2 rows in set (0.00 sec)
```

> It can be seen that pump1's state has not been updated for more than a day, the Pump is in an abnormal state, but the `State` remains' Online '. After using 'CHANGE PUMP', the Pump's `State` is changed to 'paused' :

Reviewer: ```suggestion
It can be seen that pump1's state has not been updated for more than a day, the Pump is in an abnormal state, but the `State` remains `Online`. After using `CHANGE PUMP`, the Pump's `State` is changed to 'paused' :
```
